File: Mage/UI/Theming/LightColorScheme.m
//
//  LightColorScheme.m
//  MAGE
//
//  Created by Brent Michalski on 6/5/25.
//  Copyright © 2025 National Geospatial Intelligence Agency. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "LightColorScheme.h"

@implementation LightColorScheme
- (UIColor *)primaryColor { return [UIColor systemBlueColor]; }
- (UIColor *)primaryColorVariant { return [UIColor colorWithRed:0.0 green:0.48 blue:1.0 alpha:1.0]; }
- (UIColor *)secondaryColor { return [UIColor systemOrangeColor]; }
- (UIColor *)onSecondaryColor { return [UIColor labelColor]; }
- (UIColor *)surfaceColor { return [UIColor whiteColor]; }
- (UIColor *)onSurfaceColor { return [UIColor blackColor]; }
- (UIColor *)backgroundColor { return [UIColor systemGroupedBackgroundColor]; }
- (UIColor *)onBackgroundColor { return [UIColor labelColor]; }
- (UIColor *)errorColor { return [UIColor systemRedColor]; }
- (UIColor *)onPrimaryColor { return [UIColor whiteColor]; }
@end

File: Mage/UI/Theming/Schemes/LightColorScheme.swift
//
//  LightColorScheme.swift
//  MAGE
//
//  Created by Brent Michalski on 6/9/25.
//  Copyright © 2025 National Geospatial Intelligence Agency. All rights reserved.
//

import UIKit

@objc public class LightColorScheme: NSObject, AppColorScheming {
    public var primaryColor: UIColor? = UIColor(named: "primary") ?? .systemBlue
    public var primaryColorVariant: UIColor? = UIColor(named: "primaryVariant") ?? .systemBlue
    public var secondaryColor: UIColor? = UIColor(named: "secondary") ?? .systemOrange
    public var onSecondaryColor: UIColor? = UIColor(named: "onSecondary") ?? .label
    public var surfaceColor: UIColor? = UIColor(named: "surface") ?? .systemBackground
    public var onSurfaceColor: UIColor? = .label
    public var backgroundColor: UIColor? = UIColor(named: "background") ?? .systemBackground
    public var onBackgroundColor: UIColor? = .label
    public var errorColor: UIColor? = .systemRed
    public var onPrimaryColor: UIColor? = UIColor(named: "onPrimary") ?? .white
}
